cdap-distributions/bin/build_artifact_upload.sh

```diff
@@ -82,23 +81,47 @@ function sync_build_artifacts_to_server () {
   for i in ${PACKAGES}
   do
     decho "PACKAGE=${i}"
-    _package=`echo ${i} | awk -F / '{ print $(NF) }'`
-    if [[ "${_package}" == *rpm ]]
+    _package=`basename ${i}`
+
+    ## 
+    # most CDAP Debian packages will look like this: (typical, not just in CDAP)
+    #      cdap-blah-blah-blah_2.8.0-1_all.deb
+    #      cdap-blah-blah-blah_2.8.0.1427234000548-1_all.deb
+    #    Pattern: <package name>_<version>all.deb  
+    #
+    # most CDAP RPM packages will look like this: 
+    #      cdap-blah-blah-blah-2.8.0-1.noarch.rpm
+    #      cdap-blah-blah-blah-2.8.0.1427234000548-1.noarch.rpm
+    #    Pattern: <package name>-<version>.noarch.rpm 
+    #
+    # Where version = <Major>.<Minor>.<Revision>(.optional time stamp)(-optional subrevision)
+    ##
+
+    # grab version stub
+    if [[ $_package == *_all.deb ]]
     then
-      _version_stub=`echo ${_package} | awk -F - '{ print $(NF-1) }'`
-    else
-      if [[ "${_package}" == *_* ]]
-      then
-        _version_stub=`echo ${_package} | awk -F _ '{ print $2 }'`
-      else
-        _version_stub=`echo ${_package} | awk -F - '{ print $(NF) }'`
-      fi
+      _version_stub=`echo $_package | sed 's/cdap.*_\(.*\)_all.deb/\1/'`
+    elif [[ $_package == *.noarch.rpm ]]
+    then
+      _version_stub=`echo $_package | sed 's/cdap.*-\(.*-.*\).noarch.rpm/\1/'`
+    else # send failure notification because package file has an unexpected format/pattern
+      decho "something wrong this this package"
+      die "package $_package has the wrong format: ${!}"
     fi
-    _version=`echo ${_version_stub} | awk -F . '{ print $1"."$2"."$3 }'`
+
+    # filter out version in 2 steps: 
+    #   remove trailing '-<digit(s)>' (e.g. -1)
+    #   keep just the first 3 digits in case name includes timestamp
+    decho "version stub=${_version_stub}"
+    _version=`echo ${_version_stub} | awk -F - '{ print $1 }' | awk -F . '{ print $1"."$2"."$3 }'`
     decho "version = ${_version}"
 
+    # identify and create remote incoming directory
     OUTGOING_DIR=${BUILD_PACKAGE}/${_version}
-    mkdir -p ${OUTGOING_DIR}
+    echo "Creating remote directory"
+    ssh -l ${REMOTE_USER} ${REMOTE_HOST} "sudo mkdir -p ${REMOTE_INCOMING_DIR}/${OUTGOING_DIR}" || die "could not create remote directory"
```

Comment:
Let's assume the REMOTE_USER owns this directory structure. Don't use sudo. Otherwise, you need to chown/chmod after creation. Plus, this adds the requirement that the user has sudo on the remote machine.